build-tools/packages/build-cli/src/library/repoPolicyCheck/fluidBuildTasks.ts

```diff
@@ -301,30 +231,64 @@ function eslintGetScriptDependencies(
 		throw new Error(`Unable to load eslint config file ${eslintConfig}. ${error}`);
 	}
 
-	let projects: string | string[] | undefined = config.parserOptions?.project;
-	if (projects === undefined) {
-		// If we don't have projects, our task needs to have dependent build scripts
-		return getDefaultTscTaskDependencies(root, json);
+	let projects = config.parserOptions?.project;
+	if (!Array.isArray(projects) && typeof projects !== "string") {
+		// config does not parse extends. So, "project" is what is set in top file.
+		if (projects === false || projects === null) {
+			// type based linting is disabled - assume no task prerequisites
+			return [];
+		}
+		// @typescript-eslint/parser allows true to mean use closest tsconfig.json, but we want
+		// explicit listings for dependency clarity.
+		if (projects === true) {
+			throw new Error(
+				`${json.name} eslint config 'project' is true, but should be list of tsconfig files.`,
+			);
+		}
+		// It undefined to leverage the expected base config setting of ./tsconfig.json.
+		projects = ["./tsconfig.json"];
 	}
+	const projectsArray = Array.isArray(projects) ? projects : [projects];
+
+	// Get the build scripts for the projects
+	const siblingTscScripts = projectsArray
+		// Projects with ".lint." in the name are not required to have other associated tasks.
+		.filter((project) => !project.includes(".lint."))
+		.map((project) => {
+			const found = findTscScript(json, project);
+
+			// The main compile script is build:esnext, point eslint to it
+			if (found === undefined) {
+				throw new Error(
+					`Unable to find script for project ${project} specified in ${eslintConfig}`,
```

Comment:
Probably worth being a bit more explicit about which scripts are being looked for here to make the error more acitonable.